app/src/main/java/com/example/harvestfresh/ProductListingAdapter.java

```diff
@@ -151,5 +153,18 @@ private void setAlarm() {
                     pendingIntent);
         }
 
+        private void cancelAlarm() {
+            Intent intent = new Intent(context, AlarmReceiver.class);
+
+            pendingIntent = PendingIntent.getBroadcast(context,
+                    ALARM_CODE,
+                    intent,
+                    PendingIntent.FLAG_IMMUTABLE);
+
+            if (alarmManager == null){
+                alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+            }
+            alarmManager.cancel(pendingIntent);
```

Comment:
Code safety: What if 165 returns null? Its still a good idea to nullcheck this just to be safe.